src/call/mod.rs

```diff
@@ -146,23 +268,28 @@ impl BatchContext {
     }
 
     /// Fetch the response bytes of the rpc call.
-    // TODO: return Read instead.
-    pub fn recv_message(&self) -> Option<Vec<u8>> {
-        // TODO: avoid copy
-        let len = unsafe { grpc_sys::grpcwrap_batch_context_recv_message_length(self.ctx) };
-        if len == usize::MAX {
-            return None;
-        }
-        let mut buffer = Vec::with_capacity(len);
+    pub fn recv_message(&self) -> Option<MessageReader> {
+        let buf;
+        let mut reader;
+        let length;
         unsafe {
-            grpc_sys::grpcwrap_batch_context_recv_message_to_buffer(
```

Comment:
(0): here we called a C function that does `memcpy`s inside, which copies all the received message into `buffer`. Here's the only copy that happens in the original version, it's an in place copy.